modules/class_optimize.py

```diff
@@ -373,121 +282,71 @@ def optimierung_ems(
+            ).set_startzeitpunkt(start_hour)
+            if parameter["haushaltsgeraet_dauer"] > 0
+            else None
+        )
 
+        # Initialize the inverter and energy management system
         wr = Wechselrichter(10000, akku)
-
         ems = EnergieManagementSystem(
             gesamtlast=parameter["gesamtlast"],
-            pv_prognose_wh=pv_forecast,
-            strompreis_euro_pro_wh=specific_date_prices,
+            pv_prognose_wh=parameter["pv_forecast"],
+            strompreis_euro_pro_wh=parameter["strompreis_euro_pro_wh"],
             einspeiseverguetung_euro_pro_wh=einspeiseverguetung_euro_pro_wh,
             eauto=eauto,
             haushaltsgeraet=spuelmaschine,
             wechselrichter=wr,
         )
-        o = ems.simuliere(start_hour)
-
-        ###############
-        # Optimizer Init
-        ##############
-        opti_param = {}
-        opti_param["haushaltsgeraete"] = 0
-        if spuelmaschine is not None:
-            opti_param["haushaltsgeraete"] = 1
-
-        self.setup_deap_environment(opti_param, start_hour)
-
-        def evaluate_wrapper(individual):
-            return self.evaluate(individual, ems, parameter, start_hour, worst_case)
-
-        self.toolbox.register("evaluate", evaluate_wrapper)
-        start_solution, extra_data = self.optimize(start_params)
-        best_solution = start_solution
-        o = self.evaluate_inner(best_solution, ems, start_hour)
-        eauto = ems.eauto.to_dict()
-        spuelstart_int = None
+
+        # Setup the DEAP environment and optimization process
+        self.setup_deap_environment(
+            {"haushaltsgeraete": 1 if spuelmaschine else 0}, start_hour
+        )
+        self.toolbox.register(
+            "evaluate",
+            lambda ind: self.evaluate(ind, ems, parameter, start_hour, worst_case),
+        )
+        start_solution, extra_data = self.optimize(parameter["start_solution"])
+
+        # Perform final evaluation on the best solution
+        o = self.evaluate_inner(start_solution, ems, start_hour)
         discharge_hours_bin, eautocharge_hours_float, spuelstart_int = (
-            self.split_individual(best_solution)
+            self.split_individual(start_solution)
         )
 
-        print(parameter)
-        print(best_solution)
+        # Visualize the results
         visualisiere_ergebnisse(
             parameter["gesamtlast"],
-            pv_forecast,
-            specific_date_prices,
+            parameter["pv_forecast"],
+            parameter["strompreis_euro_pro_wh"],
             o,
             discharge_hours_bin,
             eautocharge_hours_float,
-            temperature_forecast,
+            parameter["temperature_forecast"],
             start_hour,
             self.prediction_hours,
             einspeiseverguetung_euro_pro_wh,
             extra_data=extra_data,
         )
+
         os.system("cp visualisierungsergebnisse.pdf ~/")
```

Comment:
Result PDF path file should be configurable, maybe a method parameter or config parameter.